src/hypertable_insert.c

```diff
@@ -36,35 +59,35 @@ static void
 hypertable_insert_begin(CustomScanState *node, EState *estate, int eflags)
 {
 	HypertableInsertState *state = (HypertableInsertState *) node;
+	ModifyTableState *mtstate;
 	PlanState *ps;
+	bool found_cdstate = false;
+	int i;
 
 	ps = ExecInitNode(&state->mt->plan, estate, eflags);
 	node->custom_ps = list_make1(ps);
 
-	if (IsA(ps, ModifyTableState))
-	{
-		ModifyTableState *mtstate = (ModifyTableState *) ps;
-		int i;
-
-		/*
-		 * Find all ChunkDispatchState subnodes and set their parent
-		 * ModifyTableState node
-		 */
-		for (i = 0; i < mtstate->mt_nplans; i++)
-		{
-			if (IsA(mtstate->mt_plans[i], CustomScanState))
-			{
-				CustomScanState *csstate = (CustomScanState *) mtstate->mt_plans[i];
+	Assert(IsA(ps, ModifyTableState));
+	mtstate = (ModifyTableState *) ps;
 
-				if (strcmp(csstate->methods->CustomName, CHUNK_DISPATCH_STATE_NAME) == 0)
-				{
-					ChunkDispatchState *cdstate = (ChunkDispatchState *) mtstate->mt_plans[i];
+	/*
+	 * Find all ChunkDispatchState subnodes and set their parent
+	 * ModifyTableState node
+	 */
+	for (i = 0; i < mtstate->mt_nplans; i++)
+	{
+		ChunkDispatchState *cdstate =
+			(ChunkDispatchState *) get_chunk_dispatch_state(mtstate, mtstate->mt_plans[i]);
 
-					ts_chunk_dispatch_state_set_parent(cdstate, mtstate);
-				}
-			}
+		if (cdstate)
+		{
+			found_cdstate = true;
```

Comment:
This boolean seems unnecessary since it is just for assertion and you can check the pointer directly by moving the cdstate declaration outside the loop. The boolean will probably give a warning of unused variable in case assertions are off.

Reply:
The logic of that would look wrong as it would seem you would only be asserting if the last plan found a cdstate.  If you go this route, I'd recommend dropping the for loop and copying this assertion from the set_parent code:
```
	/* Inserts on hypertables should always have one subplan */
	Assert(mtstate->mt_nplans == 1);
```

Reply:
Good point @WireBaron. Then I'd keep the bool but mark it `PG_USED_FOR_ASSERTS_ONLY`